Dockerfile

```diff
@@ -34,20 +34,29 @@ RUN locale-gen en_US.UTF-8
 # configured in /etc/default/locale so we need to set it manually.
 ENV LC_ALL=en_US.UTF-8
 
-RUN adduser --gecos '' --disabled-password coder && \
+RUN addgroup --gid 1000 coder && \
+    adduser --uid 1000 --ingroup coder --home /home/coder --shell /bin/sh --disabled-password --gecos "" coder && \
 	echo "coder ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers.d/nopasswd
 
+RUN USER=coder && \
```

Comment:
use `su -c` instead or `gosu`.

Reply:
gosu is actually preferred  as it mirrors ``--user`` flag exactly and fixes weird tty and signal forwarding issues. See https://github.com/tianon/gosu. FYI Tianon creates a lot of official Docker Images on hub.docker.com.

Reply:
@sr229 @SuperSandro2000 this is setting temporary env vars for `USER` and `GROUP` as per [instructions of how to use `fixuid`](https://github.com/boxboat/fixuid#install-fixuid-in-dockerfile), which afaik is only intended for the last line in the command:

```
printf "user: $USER\ngroup: $GROUP\n" > /etc/fixuid/config.yml
```

Rather than add an additional config file externally. Has nothing to do with `gosu`, which you say to use "instead of", but there is no mention of `gosu` in the Dockerfile you're reviewing...?(unless there was in prior commit history, doesn't seem like it since the review from @sr229 is *June 18th* and last commit was *May 6th*)

Just in case there is any confusion, `fixuid` is for remapping existing file ownership of the config user:group to the given uid:gid from `--user` flag in `docker run` command. `gosu` is intended for dropping privileges from root to another user to run a command as instead:

> The core use case for `gosu` is to step *down* from `root` to a non-privileged user during container startup (specifically in the `ENTRYPOINT`, usually). - [Source](https://github.com/tianon/gosu#warning)

Reply:
@polarathene I was referring to sr229.